api/resid/gvk.go

```diff
@@ -174,17 +182,23 @@ func (x Gvk) IsSelected(selector *Gvk) bool {
 		return true
 	}
 	if len(selector.Group) > 0 {
-		if x.Group != selector.Group {
+		if !regexp.MustCompile(
```

Comment:
this approach
- compiles a regexp on each use, rather than say once at the read time of a kustomization file.
- can panic on use.
- touches all code paths that call `IsSelected` (more than just those called out in #2782).

a way to maybe fix that would be to put this stuff in
a `matchGvkByRegexp(gvk Gvk)`  method on 
https://github.com/kubernetes-sigs/kustomize/blob/master/api/types/selector.go

Compile when the instance is created.
Use it in the patch code flow.